### DISCUSSION TITLE
Is is possible to Connect one Master port to multiple Slave ports?

### DISCUSSION THREAD

**Discussion by YazhengTu** [gem5/gem5#1365] [Q&A] [answered]
I am trying to connect a master port of a ram to slave ports of multiple rams:
```
system.lvsram0.m0_port.connect(system.lvsram0.s0_ports[0])
system.lvsram0.m0_port.connect(system.lvsram1.s0_ports[0])
...
```
 but i am getting an error saying:
fatal: Port <orphan System>.lvsram0.m0_port is already connected to <orphan System>.lvsram0.s0_ports[0], cannot connect <orphan System>.lvsram1.s0_ports[0]

Is is possible doing this? Or should I use a Xbar()?

**Comment by robhau** [comment, marked as answer]
Hi @YazhengTu,

a request port (*master port* have been renamed to _request ports_ and _slave ports_ to _response ports_) can only be connected to one response port. This is also true for the inverse case.

If you take a look into the source code, you see that a request port / a response port has a private member variable of the bounded port.

https://github.com/gem5/gem5/blob/b88f814e633f879b71c636a6631f92b944017d5d/src/mem/port.hh#L134-L140

https://github.com/gem5/gem5/blob/b88f814e633f879b71c636a6631f92b944017d5d/src/mem/port.hh#L347-L353

You will need an interconnect (you already mentioned the `XBar`).
